projects/vdk-control-cli/src/vdk/internal/control/rest_lib/rest_client_errors.py

```diff
@@ -87,5 +87,18 @@ def decorated(*args, **kwargs):
                     "and points to the correct host.",
                 )
                 raise vdk_ex from ex
+            except Exception as ex:
+                log.debug(
+                    f"An gemeral Exception occurred in {fn.__module__}.{fn.__name__}",
+                    f"The Exception class was: {ex}",
+                )
+
+                vdk_ex = VDKException(
```

Comment:
We are only wrapping the original exception. Is there any benefit to do that? 

We want to avoid wrapping exceptions unless needed because that leads to a bigger logs (since now we have the original and he previous expcetion), user cannot handle it as easily (catch it). 

Also assuming all exceptions are authentication-related is certainly not correct. There might be other exceptions - bugs or the command may fail for other reason. Also they may already be hanled and formatted appropriately 

This decorator was created to catch rest API-specific exceptions and make the errors more user friendly to the user.

Reply:
I think this might work though. 

The reason no error is printed in those cases is indeed because the exception passes through  `except Exception as ex:` 

And click (the library we use) would print only errors from exceptions that inherit click.UsageError
And the authentication  exception is [VDKAuthException](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/auth_exception.py#L5)
which does not inherit click.UsageError so click prints nothing. 

Since you are wrapping it in VDKException, this would work since VDKException inherits click.UsageError 

But if the original exception  [VDKAuthException](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/auth_exception.py#L5) already contains what, why, consequences, countermeasures, and putting them again here would be ugly . 

It would be better to pass the original exception message to VDKException (we may need a new constructor  directly) without changing it.

Reply:
@antoniivanov  : Do you want me to handle the VDKAuthException ? Are we certain that the API throws in VDKAuth? Else this would get wrapped by the generic Exception. Let me know ?

Reply:
I apologize. I missed that comment. Don't be shy to spam me with reminders if I don't respond promptly enough. 

I think it would be better to handle the  catch all exceptions as well.  Because any other exception (which doesn't inherit clic.UserError) would have the same problem. So I would add a catch-all at the end as well.

Reply:
Thanks ! Have added the generatl exception along with the VDKAuthException. Kindly review.